lib/puppet/interface/action.rb

```diff
@@ -94,28 +94,34 @@ def when_rendering(type)
   # @api private
   def set_rendering_method_for(type, proc)
     unless proc.is_a? Proc
-      msg = "The second argument to set_rendering_method_for must be a Proc"
-      msg += ", not #{proc.class.name}" unless proc.nil?
+      msg = if proc.nil?
+              #TRANSLATORS 'set_rendering_method_for' and 'Proc' should not be translated
+              _("The second argument to set_rendering_method_for must be a Proc")
+            else
+              #TRANSLATORS 'set_rendering_method_for' and 'Proc' should not be translated
+              _("The second argument to set_rendering_method_for must be a Proc, not %{class_name}") %
+                  { class_name: proc.class.name }
+            end
       raise ArgumentError, msg
     end
 
     if proc.arity != 1 and proc.arity != (@positional_arg_count + 1)
-      msg =  "the when_rendering method for the #{@face.name} face #{name} action "
-      msg += "takes either just one argument, the result of when_invoked, "
-      msg += "or the result plus the #{@positional_arg_count} arguments passed "
-      msg += "to the when_invoked block, not "
-      if proc.arity < 0 then
-        msg += "a variable number"
-      else
-        msg += proc.arity.to_s
-      end
+      msg = if proc.arity < 0 then
+              #TRANSLATORS 'when_rendering', 'when_invoked' are method names and should not be translated
+              _("The when_rendering method for the %{face} face %{name} action takes either just one argument, the result of when_invoked, or the result plus the %{arg_count} arguments passed to the when_invoked block, not a variable number") %
+                  { face: @face.name, name: name, arg_count: @positional_arg_count }
+            else
+              #TRANSLATORS 'when_rendering', 'when_invoked' are method names and should not be translated
+              _("The when_rendering method for the %{face} face %{name} action takes either just one argument, the result of when_invoked, or the result plus the %{arg_count} arguments passed to the when_invoked block, not %{string}") %
+                  { face: @face.name, name: name, arg_count: @positional_arg_count, string: proc.arity.to_s }
+            end
```

Comment:
I find this 100% more readable than what it was before, only nitpick is I think we try to keep files around 80 columns, some of these lines are pretty long.

Reply:
I'm not sure I can do much about the line length its one sentence so I can't break it up further